tourneydefs.py

```diff
@@ -429,15 +429,19 @@ def update_match_scores(self):
 
     def process_game(self, scheduleid, winner_index):
         match_id = self.get_match_id_from_scheduleid(scheduleid)
-        cutoff = self.matches[match_id].best_of / 2
-        self.matches[match_id].scores[winner_index] += 1
-        self.matches[match_id].in_progress = True
-        if self.matches[match_id].scores[0] > cutoff or self.matches[match_id].scores[1] > cutoff:
-            self.matches[match_id].in_progress = False
-            self.matches[match_id].finished = True
-            self.matches[match_id].winner = winner_index
+        match = self.matches[match_id]
+        cutoff = match.best_of / 2
+        match.scores[winner_index] += 1
+        match.in_progress = True
+        if match.scores[0] > cutoff or match.scores[1] > cutoff or sum(match.scores) == match.best_of:
+            match.in_progress = False
+            match.finished = True
+            if match.scores[0] == match.scores[1]:
+                match.winner = 3  # Tie
+            else:
+                match.winner = winner_index
```

Comment:
This is actually great because an earlier version of this didn't pass by reference and I never went back and cleaned it up